independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -0,0 +1,250 @@
+                    Integer.getInteger("quarkus.analytics.timeout", DEFAULT_TIMEOUT),
+                    TimeUnit.MILLISECONDS);
+            if (log.isDebugEnabled() && !postFutures.isEmpty()) {
+                log.debug("[Quarkus build analytics] Build analytics sent successfully. Sent event can be seen at .../target/" +
+                        fileLocations.lastTrackFileName());
+            }
+        } catch (Exception e) {
+            log.warn("[Quarkus build analytics] Failed to send build analytics to Segment: " +
+                    e.getClass().getName() + ": " +
+                    (e.getMessage() == null ? "(no message)" : e.getMessage()));
+        }
+    }
+
+    List<TrackProperties.AppExtension> createExtensionsPropertyValue(ApplicationModel applicationModel) {
+        return applicationModel.getDependencies().stream()
+                .filter(dep -> dep.isResolved() &&
+                        dep.isFlagSet(TOP_LEVEL_RUNTIME_EXTENSION_ARTIFACT) &&
+                        onlyPublic(dep.getGroupId(), log))
+                .map(dep -> TrackProperties.AppExtension.builder()
+                        .groupId(dep.getGroupId())
+                        .artifactId(dep.getArtifactId())
+                        .version(dep.getVersion())
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    void sendIdentity(final Map<String, Object> context) {
+        if (this.userId.isNew()) { // fixme when inactive on 1st call it will not send identity.
+            this.restClient.postIdentity(Identity.builder()
+                    .userId(this.userId.getUuid())
+                    .context(context)
+                    .timestamp(Instant.now())
+                    .build());
+        }
+    }
+
+    Map<String, Object> createContextMap(ApplicationModel applicationModel,
+            Map<String, Object> buildInfo) {
+        WorkspaceModuleId moduleId = applicationModel.getApplicationModule().getId();
+
+        return new ContextBuilder()
+                .mapPair(PROP_APP)
+                .pair(PROP_NAME, hashSHA256(moduleId.getGroupId() + ":" + moduleId.getArtifactId()))
+                .pair(PROP_VERSION, hashSHA256(moduleId.getArtifactId() + ":" + moduleId.getVersion()))
+                .build()
+                .mapPair(PROP_JAVA)
+                .pair(PROP_VENDOR, getProperty("java.vendor", "N/A"))
+                .pair(PROP_VERSION, getProperty("java.version", "N/A"))
+                .build()
+                .mapPair(PROP_GRAALVM)
+                .pair(PROP_VENDOR, ofNullable(buildInfo.get(GRAALVM_VERSION_DISTRIBUTION)).orElse("N/A"))
+                .pair(PROP_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_VERSION)).orElse("N/A"))
+                .pair(PROP_JAVA_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_JAVA)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_BUILD)
```

Comment:
For Gradle, a couple of booleans are probably interesting as well:

* `isParallelProjectExecutionEnabled` (as `parallel`)
* `isBuildCacheEnabled` (as `build-cache`)
* `isConfigurationCacheRequested` (as `configuration-cache`)
* `isContinuous` (as `continuous`)
* `isRerunTasks` (as `rerun-tasks`)

These can be got from `Gradle.getStartParameter()`.

As a follow-up, if possible?